src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -708,7 +708,10 @@ private void CreateTypeCastPaths(ODataPath currentPath, OpenApiConvertSettings c
         /// </summary>
         private void RetrieveBoundOperationPaths(OpenApiConvertSettings convertSettings)
         {
-            foreach (var edmOperation in _model.GetAllElements().OfType<IEdmOperation>().Where(e => e.IsBound))
+            // process functions before actions
+            var edmFunctions = _model.GetAllElements().OfType<IEdmFunction>().Where(e => e.IsBound).ToList();
```

Comment:
```suggestion
            var edmFunctions = _model.GetAllElements().OfType<IEdmFunction>().Where(e => e.IsBound);
```

you're only enumerating once, no need for the to list

Reply:
Reverted.